### PR TITLE
fix(web): resolve encryption glitch in footer text 

### PR DESCRIPTION
**Pull request by Aleesssino**
## Description

I added state management to store the element's width before the encryption animation begins. This ensures that when the encryption starts, the layout remains stable and doesn't break. 
<br />

Fixes #[249](https://github.com/keyshade-xyz/keyshade/issues/249)

## Future Improvements

If others find that this fix doesn't work as expected or if the behavior isn't ideal, I will improve my solution accordingly.

## Mentions

@kriptonian1 I am waiting for your feedback.

## Screenshots of relevant screens

https://github.com/keyshade-xyz/keyshade/assets/97041873/5c5c0716-cd62-43e8-9a68-1fa0c2236d5e

## Developer's checklist

- [X] My PR follows the style guidelines of this project
- [X] I have performed a self-check on my work

**If changes are made in the code:**

- [X] I have followed the [coding guidelines](https://google.github.io/styleguide/jsguide.html)
- [X] My changes in code generate no new warnings
- [ ] My changes are breaking another fix/feature of the project
- [ ] I have added test cases to show that my feature works
- [X] I have added relevant screenshots in my PR
- [X] There are no UI/UX issues

___

### **PR Type**
Bug fix

___

### **Description**
- Added state management to store the element's width before the encryption animation begins to ensure layout stability.
- Changed the root element from `motion.button` to `motion.div` to better handle the width styling.
- Introduced a `span` with a `ref` to measure the text width and applied this width to the `motion.div`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Bug fix
</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>encrypt-text.tsx</strong><dd><code>Fix encryption animation glitch by managing element width</code></dd></summary>
<hr>

apps/web/src/components/ui/encrypt-text.tsx
<li>Added <code>useEffect</code> to store the element's width before the encryption <br>animation begins.<br> <li> Changed the root element from <code>motion.button</code> to <code>motion.div</code>.<br> <li> Added a <code>span</code> with <code>ref</code> to measure the text width.<br> <li> Applied the measured width to the <code>motion.div</code> to maintain layout <br>stability.<br>

</details>
    

  </td>
  <td><a href="https://github.com/keyshade-xyz/keyshade/pull/267/files#diff-0991949f9f3df3982b1910773d024f58ddd46c49ef2121e579fadafaf4839bce">+19/-8</a>&nbsp; &nbsp; </td>
</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: apps/web/src/components/ui/encrypt-text.tsx [hunk 1]
```diff
@@ -45,20 +51,25 @@ function EncryptText({
...
   const stopScramble = (): void => {
     clearInterval(intervalRef.current || undefined)
-
     setText(TARGET_TEXT)
   }
 
   return (
-    <motion.button
-      className=" group relative w-full overflow-hidden text-white/60 transition-colors hover:text-white"
+    <motion.div
```
[the remark this answers is not in the document]
Comment (Aleesssino) [reply apps/web/src/components/ui/encrypt-text.tsx:58]: Changing it to a button was causing unexpected behavior (see screenshot). My updated solution works as it did before.
Screenshot from 2024-06-11 20-13-05

[the remark this answers is not in the document]
Comment (kriptonian1) [reply apps/web/src/components/ui/encrypt-text.tsx:58]: Then do one thing add `role="a"` because technically it should have been a href attribute tag

**Review by kriptonian1 — APPROVED** [review]
Just add the role, and then it will be good to merge

File: apps/web/src/components/ui/encrypt-text.tsx
Comment (Aleesssino) [inline apps/web/src/components/ui/encrypt-text.tsx:1]: Updated button role to 'link' as required for accessibility.

**Review by kriptonian1 — APPROVED** [review]
@rajdip-b LGTM, you can merge 🚀

**Comment by kriptonian1** [comment]
@rajdip-b it doesn't seem like it's working after merged, can you confirm if it's an issue from the deployment end or not

**Comment by rajdip-b** [comment]
For now, we don't have any staging environment. So, the pipeline is set to work only when code is pushed to main (when a release is made). That's why, it didn't work. 

Later on when we will completely deploy the alpha environment, we will set up `develop` with the staging environment. Hope that clears things out!

**Comment by kriptonian1** [comment]
Can we merge this pr to prod

**Pull request merged**